### PR TITLE
Remove Caching from Pruner

### PR DESCRIPTION
**Pull request by TheBurchLog**
When the Pruner deletes the collection objects, they are staying cached in memory until MongoEngine de allocates them from memeory. By setting the `no_cache()` function, after the objects are iterated through (`delete()`), then drops them from memory. 

fixes beer-garden/beer-garden#604

Test Example:
120k Requests
4GB of RAM

Prior: 
Pre-Execute = 1.17GB Free
Execute = .11GB Free (lowest drop)
Post-Execute = .11GB Free

W/ Fix:
Pre-Execute = 1.18GB Free
Execute = .86GB Free (lowest drop)
Post-Execute = 1.03GB Free

### CODE REVIEW

**Review by hazmat345 — APPROVED** [review]
LGTM, but we do need to link this to an issue for the changelog.

**Comment by TheBurchLog** [comment]
We have this ticket over on the Beer-Garden Repo that I was working towards: https://github.com/beer-garden/beer-garden/issues/604

**Comment by hazmat345** [comment]
@TheBurchLog alrighty, can you edit the description to say that this PR fixes that issue?

I changed the target of this PR to be the `develop` branch. It doesn't super matter but I think we should do things the "real" way the first time :smile:

Also, the CI for this build is failing. Can you see why that's happening? We're can't release without that being fixed.

**Comment by TheBurchLog** [comment]
The CI was failing because Black was updated. Applied the same changes here as we did for the V3 branch (https://github.com/beer-garden/beer-garden/pull/585)

**Comment by hazmat345** [comment]
@TheBurchLog sounds like that wasn't related to this PR then... :upside_down_face:

Also, the build is still failing.

File: test/mongo_pruner_test.py [hunk 1]
```diff
@@ -29,4 +29,4 @@ def test_prune_something(self):
         self.mongo_pruner._stop_event = Mock(wait=Mock(side_effect=[False, True]))
 
         self.mongo_pruner.run()
-        self.assertTrue(self.collection_mock.objects.return_value.delete.called)
+        self.assertTrue(self.collection_mock.objects.return_value.no_cache.called)
```
Comment (hazmat345) [inline test/mongo_pruner_test.py:32]: We need to make sure that delete is actually called here... adding an additional check for no_cache being called is good though.

**Review by hazmat345 — CHANGES_REQUESTED** [review]

File: test/mongo_pruner_test.py (hunk 1 above)
Comment (TheBurchLog) [reply test/mongo_pruner_test.py:32]: Fixed it

**Review by hazmat345 — APPROVED** [review]
LGTM :fireworks:

**Pull request merged**